src/Mod/Path/PathScripts/post/uccnc_post.py

```diff
@@ -0,0 +1,671 @@
+			UNIT_FORMAT = UNIT_FORMAT_US_IMP
+			UNIT_SPEED_FORMAT = UNIT_SPEED_FORMAT_US_IMP
+			UNIT_DEFAULT_CHANGED = True
+
+		if args.metric and (UNITS != UNITS_METRIC):
+			print("Units: Metric [mm], check your UC-CNC profile.")
+			UNITS = UNITS_METRIC
+			UNIT_FORMAT = UNIT_FORMAT_METRIC
+			UNIT_SPEED_FORMAT = UNIT_SPEED_FORMAT_METRIC
+			UNIT_DEFAULT_CHANGED = True
+
+		if args.modal:
+			MODAL = True
+
+		if args.tool_length_offset:
+			USE_TLO = True
+
+		if args.repeat:
+			REPEAT_ARGUMENTS = True
+
+	except Exception:
+		return False
+
+	return True
+
+def append0(line):
+	result = line
+	if (trace_gcode):
+		print("export: >>" + result)
+	return result
+
+def append(line):
+	result = linenumber() + line
+	if (trace_gcode):
+		print("export: >>" + result)
+	return result
+
+def export(objectslist, filename, argstring):
+
+	if not processArguments(argstring):
+		print("export: process arguments failed, '{}'".format(argstring))
+		return None
+
+	global POST_PROCESSOR_NAME
+	global UNITS
+	global UNIT_FORMAT
+	global UNIT_SPEED_FORMAT
+	global warnings_count
+	global problems_count
+
+	warnings_count = 0
+	problems_count = 0
+
+	for obj in objectslist:
+		if not hasattr(obj, "Path"):
+			print("the object " + obj.Name + " is not a path. Please select only path and Compounds.")
+			return None
+
+	print("export: postprocessing...")
+	gcode = append0("%" + PROG_NAME + "\n")
+	if not argstring:
+		gcode += append("(" + POST_PROCESSOR_NAME + " with default settings)\n")
+	else:
+		gcode += append("({} {})\n".format(POST_PROCESSOR_NAME, argstring))
+
+	# write header
+	if OUTPUT_HEADER:
+		HEADER = '''(Exported by FreeCAD for {})
+		(Post Processor: {}, version {})
+		(CAM file: {})
+		(Output Time: {})
+		'''
+		for line in HEADER.format(MACHINE_NAME, __name__, VERSION, FreeCAD.ActiveDocument.FileName, str(now)).splitlines(True):
+			gcode += append(line)
+
+	# Write the preamble
+	# G20/G21 not supported by UC-CNC, *always* report the configured units and deviations from the default settings for safety.
+	gcode += append("(Units: '" + UNIT_FORMAT + "' and '" + UNIT_SPEED_FORMAT + "')\n")
+	if (UNIT_DEFAULT_CHANGED):
+		gcode += append("(WARNING: Units default changed, check your UC-CNC profile)\n")
+		warnings_count += 1
+
+	if OUTPUT_COMMENTS:
+		gcode += append("(preamble: begin)\n")
+		#for obj in objectslist:
+		#	if isinstance(obj.Proxy, PathScripts.PathToolController.ToolController):
+		#		gcode += append("(T{}={})\n".format(obj.ToolNumber, item.Name))
+		# error: global name 'PathScripts' is not defined
+	for line in PREAMBLE.splitlines(False):
+		gcode += append(line + "\n")
+	#if OUTPUT_COMMENTS:
+	#	gcode += append(UNITS + " (units in '" + UNIT_FORMAT + "' and '" + UNIT_SPEED_FORMAT + "')\n")
+	#else:
+	#	gcode += append(UNITS + "\n")
+	if OUTPUT_COMMENTS:
+		gcode += append("(preamble: done)\n")
+
+	# write the code body
+	for obj in objectslist:
+
+		# fetch machine details
+		job = PathUtils.findParentJob(obj)
```

Comment:
Please check this thread.  https://forum.freecadweb.org/viewtopic.php?f=15&t=56982
This is deprecated code.  The jobs no longer have the  'MachineName' or 'MachineUnits' properties so looking up the job is worthless and will lead to the deleted proxy error. 
The fix is easy and I've made it for all the other postprocessors in this PR.  https://github.com/FreeCAD/FreeCAD/pull/4673
You should make a similar fix here before this is merged.

Reply:
I have seen the topic and have your changes already in my local version. I will follow it up and push it tomorrow.

Reply:
Applied and pushed the fix.